### ISSUE TITLE
The ability for a column to be pinned

### ISSUE THREAD

**Issue by krryptonian** [icflorescu/mantine-datatable#347]
I'm typing to pin actions cells on the right side of the table but I can't do this

![image](https://github.com/icflorescu/mantine-datatable/assets/57591469/de81e5e4-afdf-4113-b1f8-d4948e0d89a9)

I have some references from the @tanstack react table here
https://tanstack.com/table/v8/docs/api/features/column-pinning

please give me the solution I am looking forward to you guys.

**Comment by krryptonian** [comment]
Looking for an replay

**Comment by SchrodingersGat** [comment]
@icflorescu I'd also be very interested in this. The left-hand "checkbox" column does this quite nicely:

![image](https://github.com/icflorescu/mantine-datatable/assets/10080325/7514fa4b-5531-42d4-ae2a-cd1e47fae2a1)

But, I cannot get the same behavior for the right hand actions menu. I can make the position fixed, and stick to the right, but it does not sit "over the top" of the other data with the nice shadow like the checkbox columns:

![image](https://github.com/icflorescu/mantine-datatable/assets/10080325/0419e823-c521-44c9-bc88-06a2f34df8a6)

**Comment by SchrodingersGat** [comment]
@krryptonian FYI I am getting pretty close with custom CSS for the "action" column:

```typescript
{
        accessor: 'actions',
        title: '   ',
        hidden: false,
        switchable: false,
        width: 50,
        cellsStyle: {
          position: 'sticky',
          right: 0,
          backgroundColor: '#FFF',
          borderLeft: '1px solid #DDD',
          padding: '3px',
        },
}
```

![image](https://github.com/icflorescu/mantine-datatable/assets/10080325/5e5a5b0d-38ac-443f-b3d8-aa8baf518873)

**Comment by SchrodingersGat** [comment]
The only missing component is how to use the ScrollArea feedback to determine when the column needs a "shadow" on the left side. @icflorescu any hints? :)

**Comment by icflorescu** [comment]
I thought about this issue from time to time, and I've reached the conclusion that adding a property called `pinLastColumn` or something like that would meet most users' expectations.

Since the title of the issue was *"The ability for a column to be pinned"*, I was initially trying to consider a way to implement pinning an arbitrary number of columns... Which would be difficult, since a table's column can scale themselves dynamically to accommodate their content. I mean, adding `sticky` and `right: 0` works for the **last** column, yes, but there's no straightforward way do determine the `left` value of the previous ones if a user would want to pin, let's say, the last 3 columns.

But if we're just talking about the last column, I think yes, it could be done in the way you are suggesting it, @SchrodingersGat. Hence my idea of the `pinLastColumn` property.

> The left-hand "checkbox" column does this quite nicely

I know, I was very fond of that feature and took the necessary time to come up with the solution. IMO it's one of those minor things pertaining to attention to details that sets Mantine DataTable apart 😁

> The only missing component is how to use the ScrollArea feedback to determine when the column needs a "shadow" on the left side.

The component triggers some callbacks you could use, like `onScrollToRight` (see https://icflorescu.github.io/mantine-datatable/examples/infinite-scrolling/), but somehow I have a feeling that implementing this feature externally would be rather cumbersome...  

Bottom line is: **I'll try to find the time to implement this internally in a few days.**

@SchrodingersGat - just curious, are the screenshots from [InvenTree](https://inventree.org/)? If so, are you planning to switch to Mantine 7? If not, I guess I'll have to backport this to V6 too... And not related to the issue - would you be ok with me adding a link to InvenTree on the Mantine DataTable docs website?

**Comment by SchrodingersGat** [comment]
> IMO it's one of those minor things pertaining to attention to details that sets Mantine DataTable apart 😁

I am *loving* working with mantine-datatable to build out the new interface for InvenTree. It has proven to be super powerful, we have built a pretty extensive [wrapper around it](https://github.com/inventree/InvenTree/blob/master/src/frontend/src/components/tables/InvenTreeTable.tsx) for rendering data straight from our API :)

> Bottom line is: I'll try to find the time to implement this internally in a few days.

Awesome! A similar visual behaviour to the checkbox column would be fantastic, that works so well!

> @SchrodingersGat - just curious, are the screenshots from [InvenTree](https://inventree.org/)? If so, are you planning to switch to Mantine 7? If not, I guess I'll have to backport this to V6 too... And not related to the issue - would you be ok with me adding a link to InvenTree on the Mantine DataTable docs website?

Yes, the screenshots above are for building out the new InvenTree UI. Currently we are focusing on v6 mantine - if you can patch against that, it would be great. We will likely upgrade to 7 but not immediately.

Please feel free to add us to your docs website 👍 we are certainly avid users of your library, that's for sure

**Comment by icflorescu** [comment]
Pinning the last column is now possible in `v7.1.4` & `v6.0.5` 🎉
See `pinLastColumn` in the docs, [V7 here](https://icflorescu.github.io/mantine-datatable/examples/pinning-the-last-column/) and [V6 here](https://icflorescu.github.io/mantine-datatable-v6/examples/pinning-the-last-column).

@SchrodingersGat, I trust the nice shadows will be to your liking 😉

**Comment by SchrodingersGat** [comment]
@icflorescu thanks for the very quick turnaround on this, it works really well!! Much appreciated :)
